### ISSUE TITLE
Use both TLS and plaintext endpoints

### ISSUE THREAD

**Issue by daniel-savu** [hyperium/tonic#1592]
### Version
`tonic = 0.9.2`

### Platform

Mac OS 13.4 

### Description

I'm trying to use tonic to connect to both tls and plaintext gRPC endpoints. I initially had a working connection to plaintext endpoints, but one of those endpoints was upgraded to tls and caused my service to crash. After following the solution in [this issue](https://github.com/cosmos/cosmos-rust/issues/361#issuecomment-1493764624), I enabled the `transport`, `tls` and `tls-roots` features and got the tls endpoint to work.

The problem is that after doing this, plaintext endpoints no longer work and return this error: `error trying to connect: received corrupt message of type InvalidContentType`.

Does the feature flagging ([here for instance](https://github.com/hyperium/tonic/blob/48dd15e7a8db644d2c5be5d43d23839f288555c5/tonic/src/transport/channel/endpoint.rs#L304)) mean that tls connections are mutually exclusive with plaintext connections?

Let me know if you have any suggestions

**Comment by LucioFranco** [comment]
Yes, I believe that is the logic if tls is enabled it does not support normal tcp connections.

**Comment by daniel-savu** [comment]
I managed to connect to plaintext endpoints in the end. My issue was caused by the fact that I was using `https` for the plaintext endpoints, when I should've used `http`. [This line](https://github.com/hyperium/tonic/blob/48dd15e7a8db644d2c5be5d43d23839f288555c5/tonic/src/transport/service/connector.rs#L71) of code is what helped me figure it out.

I'll close this issue, though let me know if this shouldn't have worked.
